core/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -857,26 +858,17 @@ public static boolean isAllOrWildcard(String[] data) {
     }
 
     /**
-     * Return a {@link String} that is the json representation of the provided
-     * {@link ToXContent}.
+     * Return a {@link String} that is the json representation of the provided {@link ToXContent}.
+     * Wraps the output into an anonymous object depending on whether the provided argument implements {@link ToXContentObject} or not
      */
     public static String toString(ToXContent toXContent) {
-        return toString(toXContent, false);
-    }
-
-    /**
-     * Return a {@link String} that is the json representation of the provided
-     * {@link ToXContent}.
-     * @param wrapInObject set this to true if the ToXContent instance expects to be inside an object
-     */
-    public static String toString(ToXContent toXContent, boolean wrapInObject) {
         try {
             XContentBuilder builder = JsonXContent.contentBuilder();
-            if (wrapInObject) {
+            if (toXContent instanceof ToXContentObject == false) {
```

Comment:
> I'd probably pull the instanceof check into a boolean variable just to make sure it is only done once.

 will do, makes sense.

> I keep thinking that we should be able to remove the instanceof check by making another method and we could do that for the most part but I don't think it'd be worth it. At least not for now, anyway.


agreed, the instanceof checks are not nice. Yet they are an improvement compared to the previous boolean flag.
If we had two methods, the method variant that accepts `ToXContent` would also accept a `ToXContentObject` which would throw error when adding the anonymous object as it already has it. Not too sure. As you said it may stay this way for now.